### PR TITLE
(GH-94) Add support for npm prune

### LINKED ISSUE #94: Add support for npm prune
would like to be able to run npm prune command using this tool.

### PR DESCRIPTION
**Pull request by rdaniels6813**
added support for npm prune

Signed-off-by: Ryan Daniels <rdaniels6813@gmail.com>

Fixes #94

### CODE REVIEW

File: src/Cake.Npm/NpmPruneAliases.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            }
+
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            AddinInformation.LogVersionInformation(context.Log);
+
+            var packer = new NpmPruneRunner(context.FileSystem, context.Environment, context.ProcessRunner, context.Tools, context.Log);
```
Comment (pascalberger) [inline src/Cake.Npm/NpmPruneAliases.cs]: Variable should be named appropriately

File: src/Cake.Npm/Prune/NpmPruneSettings.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        /// </summary>
+        public NpmPruneSettings()
+            : base("prune")
+        {
+        }
+
+        /// <summary>
+        /// Arguments to pass to npm.
+        /// </summary>
+        public IList<string> Arguments => _arguments;
```
Comment (pascalberger) [inline src/Cake.Npm/Prune/NpmPruneSettings.cs]: There's no need to provide a generic Arguments property here. There's already the [ArgumentCustomization](https://cakebuild.net/api/Cake.Core.Tooling/ToolSettings/50AAB3A8) property which can be used to add additional or manipulate arguments.

Arguments provided here should be the specific arguments as supported by npm (e.g package scope/name, production flag). But I would also be OK to take this PR without any specific arguments and add them later if there's a need.

File: src/Cake.Npm/NpmPruneAliases.cs (hunk 1 above)
Comment (rdaniels6813) [reply src/Cake.Npm/NpmPruneAliases.cs]: This has been renamed, as well as some file names I missed on the first run-through

**Comment by pascalberger** [comment]
@rdaniels6813 Your changes have been merged and will be included in upcoming 0.16.0 release. Thanks again for your contribution 👍

**Pull request merged**